### ISSUE TITLE
How to get parent field within a loop when names conflict

### ISSUE THREAD

**Issue by joshcomley** [alonrbar/easy-template-x#3]
Hi, very interested in this library, but a few questions.

Suppose this data structure:

    const data = {
        posts: [
            { author: 'Alon Bar', text: 'Very important\ntext here!', fans: [{ text: 'A' }, {text: 'B'}] },
            { author: 'Alon Bar', text: 'Forgot to mention that...', fans: [{ text: 'A' }, {text: 'B'}] }
        ]
    };

How do I fix this template:

    {#posts}
    Author: {author}
    {text}
        {#fans}
        {text} < I want the post text here
        {text} < I want the fan text here
        {/fans}
    {/posts}

Let's ignore how rubbish an example this is... but the principle remains: how would I target a the parent `text` and the new child `text` from within the second loop?

Many thanks

**Comment by alonrbar** [comment]
Hi,
Sorry for the late response, busy week...

TL;DR - currently the solution is to use different keys, for instance `postText` and `fanText`.

In general, the replacement uses "scope semantic", roughly the same way as variables scope works in js (and in [mustache](https://mustache.github.io/)). The idea was to let nested children target root properties to implement common used values (for instance a root `newPage` property with this value: `<w:br w:type="page"/>`).
You can check the code that handles data scope [here](https://github.com/alonrbar/easy-template-x/blob/master/src/compilation/scopeData.ts).
In the future it may be a good idea to support advanced paths like [handlebars ](https://handlebarsjs.com/) does, but it is not currently supported.

P.S. - if you like the package a star would be appreciated ;)

**Comment by alonrbar** [comment]
Hi,
Closing this issue.
If you think advanced path should be supported please open a specific issue for it.
